src/relay/transforms/annotate_target.cc

```diff
@@ -234,20 +236,36 @@ class AnnotateTargetRewriter : public ExprRewriter {
   Expr Rewrite_(const LetNode* op, const Expr& post) final {
     auto let = Downcast<Let>(post);
 
-    auto target_n_args = AnnotateArgs({let->value, let->body});
-    auto new_expr = Let(let->var, std::get<1>(target_n_args)[0], std::get<1>(target_n_args)[1]);
-    op_expr_to_target_[new_expr] = std::get<0>(target_n_args);
+    Expr new_expr;
+    std::pair<std::string, Array<Expr>> target_n_args;
+    Expr new_body = InsertCompilerEnd(let->body);
+    bool is_functional_literal = let->value.as<FunctionNode>() != nullptr;
+    if (is_functional_literal) {
+      new_expr = Let(let->var, let->value, new_body);
+    } else {
+      target_n_args = AnnotateArgs({let->value});
+      new_expr = Let(let->var, std::get<1>(target_n_args)[0], new_body);
+    }
+
+    return std::move(new_expr);
+  }
+
+  Expr InsertCompilerEnd(const Expr& expr) {
```

Comment:
- Put this function together with `AnnotateArgs`.
- Add formal function description.
- From its functionality, the name like `InsertCompilerEndAndPropogateTarget` would be more proper.
- Use this function in `Expr Rewrite_(const FunctionNode* fn, const Expr& post)` to avoid redundant logic.

Reply:
Done. Although using this function in AnnotateArgs makes us do the same condition check twice which I think is unnecessary. For the time being I added it, but let me know your thoughts.

Reply:
Sorry for not being clear. For the last point, I meant you can use `InertCompilerEndAndPropogateTarget` in https://github.com/apache/incubator-tvm/blob/main/src/relay/transforms/annotate_target.cc#L225 to replace the same logic.

Reply:
Ah I see , oh so thats already done now in the previous commit. I will remove it from AnnotateArgs